### PR TITLE
Rspec for price

### PR DESCRIPTION
**Pull request by sawangupta92**
1) Use constant AMOUNT_LIMIT for maximum and minimum amount.
2) Write a shared example for paranoia named “a Paranoid model”
3) Write rspecs for Included/extended modules, Constants, Associations, Validations, Callbacks, Ransackable attributes, #money, #variant, #ensure_currency
4) Move #price_including_vat_for and #display_price_including_vat_for to describe block of Instance Methods
5) Require files of shared folder in spec_helper.rb
6) Updated rspecs for #price.

### CODE REVIEW

File: core/spec/models/spree/price_spec.rb [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
 
 describe Spree::Price, :type => :model do
-  describe '#amount=' do
-    let(:price) { Spree::Price.new }
-    let(:amount) { '3,0A0' }
 
-    before do
-      price.amount = amount
-    end
+  it_behaves_like 'a Paranoid model'
```
Comment (Mafi88) [inline core/spec/models/spree/price_spec.rb]: I'm not sure we want to add these tests (this shared example basically tests gem behaviour)?
What do you think @priyank-gupta, @damianlegawiec?

File: core/spec/models/spree/price_spec.rb [hunk 2]
```diff
@@ -38,7 +41,7 @@
       it { is_expected.to be_valid }
     end
 
-    context 'when the amount is less than 0' do
+    context 'when the amount is less than #{Spree::Price::AMOUNT_LIMIT[:min]}' do
```
Comment (Mafi88) [inline core/spec/models/spree/price_spec.rb]: I think it should be " here

File: core/spec/models/spree/price_spec.rb [hunk 3]
```diff
@@ -46,83 +49,180 @@
...
+          expect(price.price_including_vat_for(tax_zone: zone)).to eq(10.00)
+        end
       end
+    end
 
-      it "returns the correct price" do
-        expect(price).to receive(:price).and_call_original
-        expect(price_with_vat).to eq(10.00)
+    describe '#display_price_including_vat_for(zone)' do
+      subject { Spree::Price.new amount: 10 }
```
Comment (Mafi88) [inline core/spec/models/spree/price_spec.rb]: Why not `build(:price, amount: 10)`?

File: core/spec/models/spree/price_spec.rb (hunk 1 above)
Comment (priyank-gupta) [inline core/spec/models/spree/price_spec.rb]: @Mafi88 Agreed. Do not see a point of testing gem behaviour.

File: core/spec/models/spree/price_spec.rb [hunk 4]
```diff
@@ -1,34 +1,37 @@
...
   end
 
-  describe '#price' do
-    let(:price) { Spree::Price.new }
-    let(:amount) { 3000.00 }
-
-    context 'when amount is changed' do
-      before do
-        price.amount = amount
+  describe 'Constants' do
```
Comment (kalkov) [inline core/spec/models/spree/price_spec.rb]: No need to test constants

File: core/spec/models/spree/price_spec.rb (hunk 2 above)
Comment (kalkov) [inline core/spec/models/spree/price_spec.rb]: Yep, that won't work.

**Pull request closed without merging**